core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -119,6 +119,16 @@ public GroupingSetDescriptor getGroupingSets()
         return groupingSets;
     }
 
+    /**
+     * Returns {@code true} if the aggregation has form GROUP BY ().
+     * Returns {@code false} if there is more than one grouping set
```

Comment:
I would remove `Returns {@code false}` case. 

I wondder why
```
even if all are empty e.g. GROUP BY GROUPING SETS ((), ()).
```
is relevant

Reply:
some places look only at `getGroupingKeys().isEmpty()` and consider this a global aggregation. This includes any number of empty grouping sets.
This method only allows for a single empty grouping set and I wanted to emphasize that in the comment so it's easier to spot the difference.

Reply:
Which places do that? That would be a bug.

There's only one possible interpretation of "single global aggregation". It must collapse all rows into a single group and produce a single row. The presence of multiple groups would immediately disqualify it as a "single" aggregation.

Reply:
> Which places do that? 

The most prominent example:
```
io.trino.sql.planner.LocalExecutionPlanner.Visitor#visitAggregation
...
         if (node.getGroupingKeys().isEmpty()) {
                return planGlobalAggregation(node, source, context);
            }
```